SubmittingPatches-backports.rst

```diff
@@ -0,0 +1,386 @@
+======================================
+Submitting Patches to Ceph - backports
+======================================
+
+Most likely you're reading this because you intend to submit a GitHub pull
+request ("PR") targeting one of the stable branches ("nautilus", etc.) at
+https://github.com/ceph/ceph.
+
+Before you open that PR, please read this entire document or, at the very least,
+the following two sections: `General principles`_ and `Cherry-picking rules`_.
+
+
+.. contents::
+   :depth: 3
+
+
+General principles
+------------------
+
+To help the people who will review your backport, please state either in the
+backport PR, or in the backport tracker issue, or in the master tracker issue:
+
+1. what bug is fixed
+2. why this fix is the minimal way to do it
+3. why does this need to be fixed in <release>
+
+The above should be followed especially in cases when the backport could be seen
+as introducing, into a stable branch, code that is not related to a particular
+bug or issue.
+
+Rationale: every modification of a stable branch carries a certain risk of
+introducing a regression. To minimize this risk, backports should be as
+straightforward and narrowly-targeted as possible. As a stable release series
+ages, the importance of following these general principles rises.
+
+
+Cherry-picking rules
+--------------------
+
+The following rules, which have been codified from "best practices" developed
+over years of backporting, apply to the actual backport implementation:
+
+* all fixes should land in master first
+* commits to stable branches should be cherry-picked from master
+* before starting to cherry-pick a set of commits from master, grep the master git history for the SHA1 of each master commit (using ``git log --grep``) to check for follow-up fixes. Include any follow-up fixes found in the set of commits to be cherry-picked.
+* cherry-picks must be done using ``git cherry-pick -x``
+* if a commit could not be cherry-picked from master, the commit message must explain why that was not possible
+* the commit message generated by ``git cherry-pick -x`` must not be modified, except to add a "Conflicts" section below the 
+* the "Conflicts" section must mention all files where changes had to made manually (not just conflicts flagged by git)
+* the "Conflicts" section should also describe the manual changes that were made
+* if a change is to be backported to multiple stable branches, a tracker issue is needed
```

Comment:
"...is needed for each branch" maybe?

Reply:
This is where folks often get confused. They think they need to manually create an issue in the Backport tracker. What is actually needed is an issue in the Bug tracker referring to the master PR and with a populated Backport field. This is explained in the `Tracker workflow` section.

Reply:
Got it. Makes sense!